### DISCUSSION TITLE
PaddleOCR/main/en/index

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleOCR#14039] [Q&A]
# PaddleOCR/main/en/index

Awesome multilingual OCR toolkits based on PaddlePaddle (practical ultra lightweight OCR system, support 80+ languages recognition, provide data annotation and synthesis tools, support training and deployment among server, mobile, embedded and IoT devices)

https://paddlepaddle.github.io/PaddleOCR/main/en/index.html

**Comment by deftydev** [comment]
When paddle ocr will be supported by new python version , is there any plan to support newer python versions?

**Reply by GreatV** [reply]
Which Python version are you referring to?

**Reply by Liyulingyue** [reply]
It my depend on the dependency of ppocr.

**Comment by PatelKartik-lgm** [comment]
Hi 
Could you please connect with your Developer or any demo person, I need brief details about this.

**Comment by Saranga99** [comment]
Anyone working with tabular data? wanted to extract different tables in a pdf file witch has around 40 pages and 50+ tables

**Reply by liuhongen1234567** [reply]
Hello, if you want to extract tables from a PDF, you can try the Table Recognition V2 pipeline at https://paddlepaddle.github.io/PaddleOCR/main/en/version3.x/pipeline_usage/table_recognition_v2.html.

**Comment by jmoore34** [comment]
Is there a library for using PaddleOCR 3.0 on Android? For example, with [Google ML Kit](https://developers.google.com/ml-kit/vision/text-recognition/v2/android) you just add the library:

```gradle
dependencies {
  implementation 'com.google.mlkit:text-recognition-chinese:16.0.1'
}
```

and then call it like so
```kt
val recognizer = TextRecognition.getClient(ChineseTextRecognizerOptions.Builder().build())
val image = InputImage.fromBitmap(bitmap, 0)
recognizer.process(image)
  .addOnSuccessListener { visionText ->
      // Task completed successfully
  }
  .addOnFailureListener { e ->
      // Task failed with an exception
  }
```
Would something similar be possible with PaddleOCR 3.0?

**Comment by BichengFan** [comment]
PP-OCRv5 检测/识别模型可以下载？

**Reply by liuhongen1234567** [reply]
您好，PP-OCRv5 检测/识别模型可以到[PP-OCRv5文档](https://paddlepaddle.github.io/PaddleOCR/main/version3.x/pipeline_usage/OCR.html#1-ocr)的模型列表中下载，如果您想要在自己数据集上微调可以下载训练模型，如果希望直接使用paddleocr wheel包推理，可以下载推理模型
<img width="1890" height="1288" alt="image" src="https://github.com/user-attachments/assets/3f792fb4-db98-4f69-89f7-34bc889ad3f5" />

**Reply by BichengFan** [reply]
使用paddlex镜像，把模型转成onnx：
python -m paddlex  --paddle2onnx  --paddle_model_dir  ./PP-OCRv5_server_det_infer  --onnx_model_dir ./PP-OCR-DET-ONNX   --opset_version 7
Input dir: PP-OCRv5_server_det_infer
Output dir: PP-OCR-DET-ONNX
Paddle2ONNX conversion starting...
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
[Paddle2ONNX] Start parsing the Paddle model file...
[Paddle2ONNX] Use opset_version = 11 for ONNX export.
[Paddle2ONNX] PaddlePaddle model is exported as ONNX format now.
2025-07-18 10:04:24 [INFO]	Try to perform constant folding on the ONNX model with Polygraphy.
[W] 'colored' module is not installed, will not use colors when logging. To enable colors, please install the 'colored' module: python3 -m pip install colored
[I] Module: 'onnxruntime' is required, but not installed. Attempting to install now.
[I] Running installation command: /usr/bin/python -m pip install onnxruntime>=1.10.0
[I] Folding Constants | Pass 1
[I] Module: 'onnx_graphsurgeon' is required, but not installed. Attempting to install now.
[I] Running installation command: /usr/bin/python -m pip install onnx_graphsurgeon>=0.3.27 --extra-index-url=https://pypi.ngc.nvidia.com
[I]     Total Nodes | Original:  1306, After Folding:   596 |   710 Nodes Folded
[I] Folding Constants | Pass 2
[I]     Total Nodes | Original:   596, After Folding:   596 |     0 Nodes Folded
2025-07-18 10:05:13 [INFO]	ONNX model saved in PP-OCR-DET-ONNX/inference.onnx.
Paddle2ONNX conversion succeeded
Copied PP-OCRv5_server_det_infer/inference.yml to PP-OCR-DET-ONNX/inference.yml
Done

使用onnxruntime(1.10.0)加载模型，
import numpy as np
import torch
import onnxruntime as ort

sess_options = ort.SessionOptions()
providers = ['CPUExecutionProvider']
detection_onnx_path = 'paddleOCR/detection/inference.onnx'
session = ort.InferenceSession(detection_onnx_path, sess_options=sess_options, providers=providers)
[ONNXRuntimeError] : 7 : INVALID_PROTOBUF : Load model from paddleOCR/detection/inference.onnx failed:Protobuf parsing failed.

请问是不是转onnx模型的时候，没成功？

**Comment by PiggyOverFlow** [comment]
请问PP-OCRV5的DB检测头，是否支持CTW-1500 或者Total-Test 数据集的训练

**Reply by liuhongen1234567** [reply]
您好，可以的，DBNet本身就是基于分割的方法，所以即使文字不规则也不影响训练。

**Reply by PiggyOverFlow** [reply]
感谢您的回复，我使用的是PP-OCRV5在下的configs/det/PP-OCRv5/PP-OCRv5_server_det.yml配置文件，当使用CTW1500或者Total-test数据集训练模型时，训练会出现卡死，详细的情况已在 issue( “https://github.com/PaddlePaddle/PaddleOCR/issues/15894”)
中描述。
期待您的回复

**Comment by AnugyaSahu** [comment]
Hi, I have a bunch of invoices in German language! and wanted to extract specific key information from them. every invoice has a different layout structure but I want the model to understand what it is searching for inside those complex structures, any ideas of what kind of pipeline should I build

**Reply by liuhongen1234567** [reply]
Hello, if you want to perform layout structure analysis, you can use the PP-Structurev3 model and then input the results of the layout analysis into large models such as GPT-4o. Alternatively, you can try using PP-ChatOCRv4, which is specifically designed for extraction tasks and has built-in large model integration. For the German language, simply replace the OCR model in the above pipeline with latin_PP-OCRv5_mobile_rec.

The relevant documentation is as follows:
 PP-Structurev3 ： https://www.paddleocr.ai/latest/en/version3.x/pipeline_usage/PP-StructureV3.html
PP-ChatOCRv4： https://www.paddleocr.ai/latest/en/version3.x/pipeline_usage/PP-ChatOCRv4.html
 latin_PP-OCRv5_mobile_rec： https://www.paddleocr.ai/latest/en/version3.x/algorithm/PP-OCRv5/PP-OCRv5_multi_languages.html

**Comment by ctgarvey** [comment]
Hi , is there perhaps a plain vanilla paddleocr 5 script and test image available for testing on my python rig? I cannot get this library  to extract any text at all, even though tesseract and easyocr can extract plenty of text. I have tried using super simple  ocr = PaddleOCR(use_angle_cls=True, lang='en') ,     result = ocr.predictor(proc)..... but it doesnt extract anything. ChatGPT gets completely confused with the paddleocr options, mixing up all the versions, and cannot generate anything that works. Is thre a limit on image size? i was trying with image (3512, 4968, 3). Many thanks!

**Reply by liuhongen1234567** [reply]
Hello, since (3512, 4968, 3) is a relatively large size and exceeds the default training configuration, you can resize the image to within 960, for example:
```python
ocr = PaddleOCR(use_angle_cls=True, lang='en',text_det_limit_type='max',text_det_limit_side_len=960)
```
For more parameters, you can refer to the PP-OCRv5 documentation.
https://www.paddleocr.ai/latest/en/version3.x/pipeline_usage/OCR.html

**Comment by richygithub** [comment]
使用PaddleOCR识别图片中的文字时，发现文字的位置和原图片有偏移，是否可以获取文字在原图的位置？

**Comment by jeremyxie97** [comment]
你好，请问 paddleocr-vl-1.5 会提供稳定的 API 服务吗？有业务需求会用到。
以及后续会支持返回带角度的识别结果吗？有实际需求场景会用到。感谢回复～

**Reply by liuhongen1234567** [reply]
您好，稳定 api 服务 可以移步到百度智能云，文档如下： https://cloud.baidu.com/product/ocr/doc_parser。免费api 可以去paddleocr 官网 https://aistudio.baidu.com/paddleocr。通过指定 layout_shape_mode=poly 检测多边形框，可以适用于任意形状，也涵盖旋转后的各种文档，并没有计划返回角度。

**Comment by aameliig** [comment]
Hello! I use eslav_PP-OCRv5_mobile_rec to recognize words in russian language. I have a lot of recognition mistakes, though.

The most common are:
- incorrect or missing letters
- replacement cyrillic characters with english
- transliteration ( "вич" in russian as "vich" in english, same sounds)
- absence of space characters

An example:
Ground Truth: АЛИНА
Model: АНА

Surprisingly, this happens even while processing a good-quality photo. 

Please can you help me to figure out the problem of such kind of mistakes? Maybe I need to use a model in a specific way to get a better result?
